### ISSUE TITLE
Bug with symbol keys in partial templates completely breaking iterators

### ISSUE THREAD

**Issue by Krastanov** [jverzani/Mustache.jl#139]
This works:

```julia
A = [Dict("a" => "eh", "b" => "bee"),
     Dict("a" => "ah", "b" => "buh")]
tpl = mt"{{#:A}}Pronounce a as {{>:d}} and b as {{b}}. {{/:A}}"
render(tpl, A=A, d="*{{a}}*") |> print
# Pronounce a as *eh* and b as bee. Pronounce a as *ah* and b as buh. 
```

This produces very peculiar repeated wrong strings:

```julia
A = [Dict(:a => "eh", :b => "bee"),
     Dict(:a => "ah", :b => "buh")]
tpl = mt"{{#:A}}Pronounce a as {{>:d}} and b as {{:b}}. {{/:A}}"
render(tpl, A=A, d="*{{:a}}*") |> print
# Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . Pronounce a as ** and b as . 
 ```

**Comment by Krastanov** [comment]
Related, when I use named tuples instead of dicts, another problem appears:

```julia
A = [(a = "eh", b = "bee"),
     (a = "ah", b = "buh")]
tpl = mt"{{#:A}}Pronounce a as {{>:d}} and b as {{:b}}. {{/:A}}"
render(tpl, A=A, d="*{{:a}}*") |> print
```
raises
```
ERROR: LoadError: MethodError: no method matching occursin(::Regex, ::Vector{Any})
Closest candidates are:
  occursin(::Regex, ::SubString; offset) at regex.jl:269
  occursin(::Regex, ::AbstractString; offset) at regex.jl:264
  occursin(::Any) at strings/search.jl:636
Stacktrace:
  [1] renderTokens(io::IOBuffer, tokens::Vector{Any}, writer::Mustache.Writer, context::Mustache.Context, template::Mustache.MustacheTokens, idx::Tuple{Int64, Int64})
    @ Mustache ~/.julia/packages/Mustache/xLZ0S/src/tokens.jl:649
  [2] renderTokensByValue(value::Vector{NamedTuple{(:a, :b), Tuple{String, String}}}, io::IOBuffer, token::Mustache.SectionToken, writer::Mustache.Writer, context::Mustache.Context, template::Mustache.MustacheTokens, args::Tuple{Int64, Int64})
    @ Mustache ~/.julia/packages/Mustache/xLZ0S/src/tokens.jl:462
  [3] renderTokens(io::IOBuffer, tokens::Vector{Mustache.Token}, writer::Mustache.Writer, context::Mustache.Context, template::Mustache.MustacheTokens, idx::Tuple{Int64, Int64})
    @ Mustache ~/.julia/packages/Mustache/xLZ0S/src/tokens.jl:600
  [4] renderTokens
    @ ~/.julia/packages/Mustache/xLZ0S/src/tokens.jl:592 [inlined]
  [5] f
    @ ~/.julia/packages/Mustache/xLZ0S/src/writer.jl:46 [inlined]
  [6] render(io::IOBuffer, w::Mustache.Writer, template::Mustache.MustacheTokens, view::Dict{Symbol, Any})
    @ Mustache ~/.julia/packages/Mustache/xLZ0S/src/writer.jl:53
  [7] render
    @ ~/.julia/packages/Mustache/xLZ0S/src/render.jl:23 [inlined]
  [8] (::Mustache.var"#14#15"{Base.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:A, :d), Tuple{Vector{NamedTuple{(:a, :b), Tuple{String, String}}}, String}}}, Mustache.MustacheTokens})(io::IOBuffer)
    @ Mustache ~/.julia/packages/Mustache/xLZ0S/src/render.jl:30
  [9] sprint(::Function; context::Nothing, sizehint::Int64)
    @ Base ./strings/io.jl:112
 [10] sprint
    @ ./strings/io.jl:106 [inlined]
 [11] #render#13
    @ ~/.julia/packages/Mustache/xLZ0S/src/render.jl:30 [inlined]
```

**Comment by jverzani** [comment]
Thanks for letting me know!

**Comment by jverzani** [comment]
Thanks again! This exposed a mishandling of Tables.jl data with partials, of which the symbols were part of. Hopefully this is fixed with #140, if you find other issues, please share.

**Comment by Krastanov** [comment]
Thank you! This is great!
